src/settings.js

```diff
@@ -1453,6 +1453,23 @@ var USE_PTHREADS = 0;
 // [link] - affects generated JS runtime code at link time
 var PTHREAD_POOL_SIZE = '';
 
+// Normally, applications can create new threads even when the pool is empty.
+// When application breaks out to the JS event loop before trying to block on
```

Comment:
To me, "break out" seems closer to what should happen (since you need to essentially bypass any normal control flow with either Emscripten's event loop APIs, or Asyncify or something else), but I don't feel strongly about it.